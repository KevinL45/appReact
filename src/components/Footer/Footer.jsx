//Import le css de Footer
import style from "./Footer.module.css"

const Footer = () =>{
    return (
        <footer>
            <div className={style.footer}>
                <h3>Menus</h3>
                <ul>
                    <li>Accueil</li>
                    <li>Galerie</li>
                    <li>A propos</li>
                    <li>Contact</li>
                </ul>
            </div>
            <div className={style.footer}>
                <h3>RÃ©seaux sociaux</h3>
                <ul>
                    <li>Twitter</li>
                    <li>Youtube</li>
                </ul>
            </div>
            <div className={style.footer}>
                <h3>Besoin d'information ?</h3>
                <ul>
                    <li>frocusx@hotmail.fr</li>
                </ul>
            </div>
        </footer>

    )

    
}

//Permet d'exporter le composant Footer 
export default Footer;