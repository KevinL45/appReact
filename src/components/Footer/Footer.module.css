footer{
    background-color: #f3f3f3;    /* Couleur d'arrière-plan */
    padding:5px;
    text-align: center;
}
.piedpage{

    text-align: center;


}
.grille{
    display: grid;
    grid-template-columns: repeat(4, 300px);
    
}

.droit {
    border-top: 1px solid black;       /* Ajout d'une bordure */
    padding:5px;
    font-family: Arial;


}


.onglets{
    list-style: none;
    text-decoration: none;        /* Suppression du soulignement */
    border-radius: 4px;           /* Arrondis des bordures */
    font-family: Arial;
    width: 180px;             /* Largeur minimale des liens */  
    font-family: Arial;
    font-weight: bold;


    
}


.lien {
    
    border: 1px solid #fff;       /* Ajout d'une bordure */
    border-radius: 4px;           /* Arrondis des bordures */
    background-color: #fff;    /* Couleur d'arrière-plan */
    padding:5px;
    text-align: center;
}

.titre {
    
    padding:5px;
    text-align: center; 
    border-bottom: 3px solid black;       /* Ajout d'une bordure */

}


