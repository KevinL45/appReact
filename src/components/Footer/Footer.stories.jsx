import Footer from "./Footer";

export default {
    title: "Components/Footer",
    components : Footer
}

export const Piedpage = () => <Footer/>
