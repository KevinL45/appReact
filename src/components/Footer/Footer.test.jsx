//Importe le composant Footer 
import Footer from "./Footer";
//Importe render et screen dans library
import {render, screen} from "@testing-library/react"

//Test
test("Test", ()=>{
const rendered = render(<Footer/>);

//Effet visuel
expect(rendered).toMatchSnapshot();
}
)