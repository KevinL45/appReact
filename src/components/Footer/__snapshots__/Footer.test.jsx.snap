// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer>
        <div
          class="grille"
        >
          <div>
            <ul
              class="onglets"
            >
              <li
                class="titre"
              >
                MENU
              </li>
            </ul>
            <div
              class="ligne"
            />
            <ul
              class="onglets"
            >
              <li
                class="lien"
              >
                Accueil
              </li>
              <li
                class="lien"
              >
                Galerie
              </li>
              <li
                class="lien"
              >
                A propos
              </li>
              <li
                class="lien"
              >
                Contact
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="onglets"
            >
              <li
                class="titre"
              >
                RESEAUX SOCIAUX
              </li>
            </ul>
            <div
              class="ligne"
            />
            <ul
              class="onglets"
            >
              <li
                class="lien"
              >
                Twitter
              </li>
              <li
                class="lien"
              >
                Youtube
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="onglets"
            >
              <li
                class="titre"
              >
                ME CONTACTER
              </li>
            </ul>
            <div
              class="ligne"
            />
            <ul
              class="onglets"
            >
              <li
                class="lien"
              >
                frocusx@hotmail.fr
              </li>
            </ul>
          </div>
          <div>
            <ul
              class="onglets"
            >
              <li
                class="titre"
              >
                LOGO
              </li>
            </ul>
            <div
              class="ligne"
            />
            <ul
              class="onglets"
            >
              <li
                class="lien"
              >
                En cours...
              </li>
            </ul>
          </div>
        </div>
        <div
          class="droit"
        >
          © 2020 - Tous droits réservés
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer>
      <div
        class="grille"
      >
        <div>
          <ul
            class="onglets"
          >
            <li
              class="titre"
            >
              MENU
            </li>
          </ul>
          <div
            class="ligne"
          />
          <ul
            class="onglets"
          >
            <li
              class="lien"
            >
              Accueil
            </li>
            <li
              class="lien"
            >
              Galerie
            </li>
            <li
              class="lien"
            >
              A propos
            </li>
            <li
              class="lien"
            >
              Contact
            </li>
          </ul>
        </div>
        <div>
          <ul
            class="onglets"
          >
            <li
              class="titre"
            >
              RESEAUX SOCIAUX
            </li>
          </ul>
          <div
            class="ligne"
          />
          <ul
            class="onglets"
          >
            <li
              class="lien"
            >
              Twitter
            </li>
            <li
              class="lien"
            >
              Youtube
            </li>
          </ul>
        </div>
        <div>
          <ul
            class="onglets"
          >
            <li
              class="titre"
            >
              ME CONTACTER
            </li>
          </ul>
          <div
            class="ligne"
          />
          <ul
            class="onglets"
          >
            <li
              class="lien"
            >
              frocusx@hotmail.fr
            </li>
          </ul>
        </div>
        <div>
          <ul
            class="onglets"
          >
            <li
              class="titre"
            >
              LOGO
            </li>
          </ul>
          <div
            class="ligne"
          />
          <ul
            class="onglets"
          >
            <li
              class="lien"
            >
              En cours...
            </li>
          </ul>
        </div>
      </div>
      <div
        class="droit"
      >
        © 2020 - Tous droits réservés
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
